### PR TITLE
Fix Windows signals incompatibilities

### PR DESCRIPTION
**Pull request by pashagolub**
On Windows only SIGTERM, CTRL_C_EVENT and CTRL_BREAK_EVENT signals are supported. However, sending Ctrl+Break requires already being attached to the same console as the target. Note that the target isn't necessarily a single process that's attached to the console. Sending a control event is implemented by calling WinAPI GenerateConsoleCtrlEvent, which targets process group identifiers (i.e. like Unix killpg), not process identifiers (i.e. not like Unix kill). 

So the only sane solution is to send SIGTERM for all modes supported.

### CODE REVIEW

**Comment by CyberDem0n** [comment]
Yeah, Windows is a quite special beast...
Your change doesn't seem right to me. The SIGTERM is used for the smart shutdown, which is never triggered by Patroni. We are using only fast and immediate.
I just checked the source code and figured out that pg_ctl on Windows is using a special PIPE to send kind of unix signals to a postmaster. In order to get everything working correctly on Windows, we have to use `pg_ctl kill    SIGNALNAME PID` instead of just calling `send_signal`, which is in turn is a simple wrapper of `os.kill()`.

File: patroni/postgresql/postmaster.py [hunk 1]
```diff
@@ -102,15 +108,23 @@ def signal_stop(self, mode):
...
+            try:
+                self.send_signal(STOP_SIGNALS[mode])
+                return None
+            except psutil.NoSuchProcess:
+                return True
+            except psutil.AccessDenied as e:
+                logger.warning("Could not send stop signal to PostgreSQL (error: {0})".format(e))
+                return False
+        else:
+            if subprocess.call(['pg_ctl', 'kill', STOP_SIGNALS[mode], self.pid]) == 0:
```
Comment (CyberDem0n) [inline patroni/postgresql/postmaster.py]: Usually `pg_ctl` is not in the default path, you have to use [pg_ctl method](https://github.com/zalando/patroni/blob/master/patroni/postgresql/__init__.py#L160-L166) instead.
Another problem is with passing `self.pid`, which is an integer, while command-line arguments could only be strings.

File: patroni/postgresql/postmaster.py [hunk 2]
```diff
@@ -123,7 +124,7 @@ def signal_stop(self, mode):
             elif not self.is_running():
                 return True
             else:
-                logger.warning("Could not send stop signal %s to postmaster with PID %s", STOP_SIGNALS[mode], self.pid)
+                logger.warning("Could not send stop signal to PostgreSQL with PID {0}".format(self.pid))
```
Comment (CyberDem0n) [inline patroni/postgresql/postmaster.py]: Why this change? `%s` formatting is preferred with the logging module

File: patroni/postgresql/postmaster.py (hunk 1 above)
Comment (pashagolub) [reply patroni/postgresql/postmaster.py]: But `pg_ctl kill` cannot have `['-D', self._data_dir]`, but they are hardcoded in pg_ctl method. Rewrite it?

Comment (CyberDem0n) [reply patroni/postgresql/postmaster.py]: Oh, You are right.

Than `subprocess.call(['pg_ctl', 'kill', STOP_SIGNALS[mode], self.pid])` should be changed to:
```
subprocess.call([pgcommand('pg_ctl'), 'kill', STOP_SIGNALS[mode], str(self.pid)])
```

Or maybe just pass `pg_ctl` as an argument to the `signal_stop()`. It is kind of ugly, but we already doing it with `PostmasterProcess.start()`

File: patroni/postgresql/postmaster.py (hunk 2 above)
Comment (pashagolub) [reply patroni/postgresql/postmaster.py]: To be consistent with the similar log message several lines before :)

Comment (CyberDem0n) [reply patroni/postgresql/postmaster.py]: I would actually like to have the other line fixed too, for consistency :)
